semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -1241,6 +1244,80 @@ and m_other_argument_operator = m_other_xxx
 
 (*e: function [[Generic_vs_generic.m_other_argument_operator]] *)
 
+(*---------------------------------------------------------------------------*)
+(* Associative-commutative iso *)
+(*---------------------------------------------------------------------------*)
+and m_call_op aop aargs bop tokb bargs =
+  let m_op_default =
+    m_arithmetic_operator aop bop >>= fun () -> m_arguments aargs bargs
+  in
+  (* We first try to perform AC-matching if possible, otherwise we fallback
+   * to the default matching strategy for non-AC operators. *)
+  match
+    ( H.ac_matching_nf aop (A.unbracket aargs),
+      H.ac_matching_nf bop (B.unbracket bargs) )
+  with
+  | Some aargs_ac, Some bargs_ac when aop =*= bop ->
+      if_config
+        (fun x -> x.ac_matching)
+        ~then_:(m_ac_op tokb aop aargs_ac bargs_ac)
+        ~else_:m_op_default
+  | ___else___ -> m_op_default
+
+(* Associative-Commutative (AC) matching of operators.
+ *
+ * This for example, will successfully match `a && b && c` against `b && a && c`!
```

Comment:
maybe you can add that "
this will also let match `if (<... b && c ...>) `  against `if(a && b && c)` (which internally was parsed
 as `if ((a && b) && c)` which was not matching before.